activerecord/test/cases/associations/eager_test.rb

```diff
@@ -209,6 +209,14 @@ def test_type_cast_in_where_references_association_name
     assert_equal [child], comment.children
   end
 
+  def test_eager_with_conditional_on_the_association
+    welcome = posts(:welcome)
+    posts = Post.includes(:categories).where(id: welcome.id, categories: { id: 1 })
+    assert_equal 2, welcome.categories.size
+    assert_equal 1, posts.size
+    assert_equal 1, posts[0].categories.size
+  end
+
```

Comment:
I think we already test this, it just uses an older notation.
For example:
https://github.com/rails/rails/blob/6d22613bbeaac258e75ea658ff709cc5feeaf121/activerecord/test/cases/associations/eager_test.rb#L1188-L1204

Reply:
@p8  Thanks for pointing that out. I've removed the new test from this PR so now the PR only contains the update to the documentation to make the behaviour of `includes` when used with a conditional, more clear.